source/javascripts/controllers/_StackController.js.erb

```diff
@@ -87,6 +110,31 @@
 				return stack.name || stack.id;
 			};
 
+			// NOTE: Conditionally show the selector if the user is the owner and has platform admin beta tag and the the pool is loaded
+			viewModel.canShowStackSelector = function() {
+				if(viewModel.isPlatformAdmin && !appService.agentPool) {
+						return false;
+				}
+				return true;
+			}
+
+			viewModel.canShowDefaultInvalidSelectorError = function() {
+				var stacks = Stack.all;
+				if(viewModel.isPlatformAdmin && appService.agentPool) {
+					stacks = stacks.concat(viewModel.agentPoolStacks);
+				}
+				return stacks.indexOf(appService.stack) == -1
+			}
+
+			viewModel.canShowInvalidStackError = function(workflow) {
+				var stacks = Stack.all;
+				if(viewModel.isPlatformAdmin && appService.agentPool) {
+					stacks = stacks.concat(viewModel.agentPoolStacks);
+				}
+				var stack = viewModel.stackGetterSetterForWorkflow(workflow, true)()
+				return stacks.indexOf(stack) == -1
+			}
+
```

Comment:
If the agent pool stacks are also in `Stack.all`, then these methods are not needed, because the `Stack.all.indexOf(...)` will be sufficient in the template.